src/main/java/frc/robot/subsystems/swerve/states/SwerveStateHelper.java

```diff
@@ -0,0 +1,75 @@
+package frc.robot.subsystems.swerve.states;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import frc.constants.MathConstants;
+import frc.robot.subsystems.swerve.Swerve;
+import frc.robot.subsystems.swerve.SwerveConstants;
+import frc.robot.subsystems.swerve.module.ModuleUtils;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class SwerveStateHelper {
+
+	private final Swerve swerve;
+	private final SwerveConstants swerveConstants;
+	private final Supplier<Optional<Pose2d>> robotPoseSupplier;
+	private final Supplier<Optional<Translation2d>> objectTranslationSupplier;
+
+	public SwerveStateHelper(
+		Supplier<Optional<Pose2d>> robotPoseSupplier,
+		Supplier<Optional<Translation2d>> objectTranslationSupplier,
+		Swerve swerve
+	) {
+		this.swerve = swerve;
+		this.swerveConstants = swerve.getConstants();
+		this.robotPoseSupplier = robotPoseSupplier;
+		this.objectTranslationSupplier = objectTranslationSupplier;
+	}
+
+	public ChassisSpeeds applyAimAssistOnChassisSpeeds(ChassisSpeeds speeds, SwerveState swerveState) {
+		return switch (swerveState.getAimAssist()) {
+			case NONE -> speeds;
+		};
+	}
+
+	public Translation2d getRotationAxis(RotateAxis rotationAxisState) {
+		return switch (rotationAxisState) {
+			case MIDDLE_OF_CHASSIS -> new Translation2d();
+			case FRONT_LEFT_MODULE -> swerve.getModules().getModule(ModuleUtils.ModulePosition.FRONT_LEFT).getPositionFromCenterMeters();
+			case FRONT_RIGHT_MODULE -> swerve.getModules().getModule(ModuleUtils.ModulePosition.FRONT_RIGHT).getPositionFromCenterMeters();
+			case BACK_LEFT_MODULE -> swerve.getModules().getModule(ModuleUtils.ModulePosition.BACK_LEFT).getPositionFromCenterMeters();
+			case BACK_RIGHT_MODULE -> swerve.getModules().getModule(ModuleUtils.ModulePosition.BACK_RIGHT).getPositionFromCenterMeters();
+		};
+	}
+
+	public RotateAxis getFarRotateAxis(boolean isLeft) {
+		Rotation2d currentAllianceHeading = swerve.getAllianceRelativeHeading();
```

Comment:
```suggestion
		Rotation2d currentHeadingRelativeToAlliance = swerve.getAllianceRelativeHeading();
```
It has nothing to do with the heading of the alliance itself (whatever that means)